setup.py

```diff
@@ -662,6 +663,25 @@ def save_build_and_package_info(package_name, version_number, cuda_version, rocm
 
     save_build_and_package_info(package_name, version_number, cuda_version, rocm_version)
 
+    def set_training_torch_interop_enabler():
```

Comment:
Instead of doing this file parsing and function replacement hack, can we instead set a `constexpr bool` in pybind to `true` when ort is built with `--enable_training_torch_interop` and to `false` when ort is not built with `--enable_training_torch_interop`.

Then, we can just call `enable_custom_autograd_support(<constexpr_value_from_pybind>)`.

Reply:
Interesting idea. So where we should store the true or false, during build, we need a persistent store for this flag any way, e.g. setting the value in memory in build time (e.g. --enable_training_torch_interop), and retrieving the value in training runtime.